File: Test/TestProto.cpp
#include <iostream>
#include <fstream>
#include "../Proto/RDPBoost.pb.h"

int main()
{
    std::ifstream readstream("savefile");
    char Readbuffer[256] = "";
    readstream.read(Readbuffer, sizeof(Readbuffer));
    std::cout<<"read count:"<<readstream.gcount()<<std::endl;

    ProtoMessage recvMessage;
    if(!recvMessage.ParseFromArray(static_cast<void*>(Readbuffer), readstream.gcount()))
    {
        std::cout<<"parse message error"<<std::endl;
        return 0;
    }

    const VideoMessage& videoMessageI = recvMessage.videmessagei();
    std::cout<<videoMessageI.width()<<std::endl;
}

File: Test/TestProto.py
import RDPBoost_pb2

data = b'123'

CSendMessage = RDPBoost_pb2.ProtoMessage()
CSendMessage.type =  RDPBoost_pb2.ProtoMessage.DataType.VIDEO_MESSAGE
CSendMessage.VideMessageI.width = 1920
CSendMessage.VideMessageI.height = 1080
CSendMessage.VideMessageI.frame_rate = 30
CSendMessage.VideMessageI.bit_rate = 4000000
CSendMessage.VideMessageI.data = data
sendata = CSendMessage.SerializeToString()

with open('savefile','wb') as fd:
    fd.write(sendata)